pkg/capabilities/consensus/ocr3/aggregators/reduce_test.go

```diff
@@ -0,0 +1,928 @@
+				},
+				shouldReport: true,
+				expectedOutcome: map[string]any{
+					"Reports": []any{
+						map[string]any{
+							"Price": decimal.NewFromInt(32),
+						},
+					},
+				},
+				expectedState: map[string]any{
+					"BenchmarkPrice": decimal.NewFromInt(32),
+				},
+			},
+			{
+				name: "float64 median",
+				fields: []aggregators.AggregationField{
+					{
+						InputKey:        "BenchmarkPrice",
+						OutputKey:       "Price",
+						Method:          "median",
+						DeviationString: "10",
+						DeviationType:   "percent",
+					},
+				},
+				extraConfig: map[string]any{},
+				observationsFactory: func() map[commontypes.OracleID][]values.Value {
+					mockValue, err := values.WrapMap(map[string]any{
+						"BenchmarkPrice": float64(1.2),
+					})
+					require.NoError(t, err)
+					return map[commontypes.OracleID][]values.Value{1: {mockValue}, 2: {mockValue}, 3: {mockValue}}
+				},
+				shouldReport: true,
+				expectedOutcome: map[string]any{
+					"Reports": []any{
+						map[string]any{
+							"Price": float64(1.2),
+						},
+					},
+				},
+				expectedState: map[string]any{
+					"BenchmarkPrice": float64(1.2),
+				},
+			},
+			{
+				name: "time median",
+				fields: []aggregators.AggregationField{
+					{
+						InputKey:        "BenchmarkPrice",
+						OutputKey:       "Price",
+						Method:          "median",
+						DeviationString: "10",
+						DeviationType:   "percent",
+					},
+				},
+				extraConfig: map[string]any{},
+				observationsFactory: func() map[commontypes.OracleID][]values.Value {
+					mockValue, err := values.WrapMap(map[string]any{
+						"BenchmarkPrice": now,
+					})
+					require.NoError(t, err)
+					return map[commontypes.OracleID][]values.Value{1: {mockValue}, 2: {mockValue}, 3: {mockValue}}
+				},
+				shouldReport: true,
+				expectedOutcome: map[string]any{
+					"Reports": []any{
+						map[string]any{
+							"Price": time.Time(now).UTC(),
+						},
+					},
+				},
+				expectedState: map[string]any{
+					"BenchmarkPrice": now.UTC(),
+				},
+			},
+			{
+				name: "big int median",
+				fields: []aggregators.AggregationField{
+					{
+						InputKey:        "BenchmarkPrice",
+						OutputKey:       "Price",
+						Method:          "median",
+						DeviationString: "10",
+						DeviationType:   "percent",
+					},
+				},
+				extraConfig: map[string]any{},
+				observationsFactory: func() map[commontypes.OracleID][]values.Value {
+					mockValue, err := values.WrapMap(map[string]any{
+						"BenchmarkPrice": big.NewInt(100),
+					})
+					require.NoError(t, err)
+					return map[commontypes.OracleID][]values.Value{1: {mockValue}, 2: {mockValue}, 3: {mockValue}}
+				},
+				shouldReport: true,
+				expectedOutcome: map[string]any{
+					"Reports": []any{
+						map[string]any{
+							"Price": big.NewInt(100),
+						},
+					},
+				},
+				expectedState: map[string]any{
+					"BenchmarkPrice": big.NewInt(100),
+				},
+			},
+			{
+				name: "buffer mode",
```

Comment:
what is buffer mode and how would this work if the method was `median`? is that valid?

Reply:
Finding a mode on a data set of buffer / []byte types. Taking `median` of buffers isn't valid, because you can't sort them using a numeric comparison (`>`). We could support them by sorting them in another way though.

Reply:
Renamed to bytes to make it more clear